subgraphs/uniswap-v2/src/common/helpers.ts

```diff
@@ -0,0 +1,518 @@
+
+  // Get new tvl
+  let newTvl = tokenTracker0.derivedETH.times(inputToken0).times(ether.valueDecimal!).plus(tokenTracker1.derivedETH.times(inputToken1).times(ether.valueDecimal!))
+
+  // Add the new pool tvl
+  pool.totalValueLockedUSD =  newTvl
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.plus(newTvl)
+
+  let outputTokenSupply = convertTokenToDecimal(pool.outputTokenSupply, DEFAULT_DECIMALS)
+
+  // Update LP token prices
+  if (pool.outputTokenSupply == BIGINT_ZERO) pool.outputTokenPriceUSD = BIGDECIMAL_ZERO
+  else pool.outputTokenPriceUSD = newTvl.div(outputTokenSupply)
+
+
+  pool.save()
+  protocol.save()
+  ether.save()
+}
+
+// Checks if the mint has been completed with the Deposit event. 
+function isCompleteMint(mintId: string): boolean {
+  let mint = getMint(mintId)
+  if (mint == null) return false
+  return mint.from !== null // sufficient checks
+}
+
+// Handle data from transfer event for mints. Used to populate deposit entity in the mint event. 
+export function handleTransferMint(event: ethereum.Event, value: BigInt, to: Address): void {
+  let pool = getLiquidityPool(event.address.toHexString())
+  let transaction = getOrCreateTransaction(event)
+  
+  // Tracks supply of minted LP tokens 
+  pool.outputTokenSupply = pool.outputTokenSupply.plus(value)
+  pool.save()
+
+  let mints = transaction.mints
+
+  // create new mint if no mints so far or if last one is done already
+  if (mints == null || isCompleteMint(mints[mints.length - 1])) {
+    
+    let length = mints == null ? 0: mints.length
+
+    let mint = createMint(event, length)
+
+    mint.to = to
+    mint.liquidity = value
+
+    // update mints in transaction
+    if (mints == null) transaction.mints = [mint.id]
+    else transaction.mints = mints.concat([mint.id])
+
+    // save entities
+    transaction.save()
+    mint.save()
+  }
+
+  // This is done to remove a potential feeto mint --- Not active 
+  else if (!isCompleteMint(mints[mints.length - 1])) {
+    store.remove('Mint', mints[mints.length - 1])
+
+    let mint = createMint(event, mints.length)
+
+    mint.to = to
+    mint.liquidity = value
+  
+    // update mints in transaction
+    mints.pop()
+    transaction.mints = mints.concat([mint.id])
+  
+    // save entities
+    transaction.save()
+    mint.save()
+  }
+}
+
+// Handle the transer of LP tokens to the liquidity pool that minted them and track the data for the burn event. 
+export function handleTransferToPool(event: ethereum.Event, value: BigInt, to: Address, from: Address): void {
+  let transaction = getOrCreateTransaction(event)
+
+  let burns = transaction.burns
+  let length = burns == null ? 0: burns.length
+  let burn = createBurn(event, length)
+
+  burn.liquidity = value
+  burn.to = to
+  burn.sender = from
+  burn.needsComplete = true
+
+  // update burns in transaction
+  if (burns == null) transaction.burns = [burn.id]
+  else transaction.burns = burns.concat([burn.id])
```

Comment:
Why are we grouping multiple burns in the same transaction into a Transaction entity?

Reply:
As mentioned - I deleted a bunch of code and simplified this process.  This is gone.